nfuzz/libnfuzz.nim

```
@@ -84,51 +92,61 @@ proc nfuzz_attester_slashing(input: openArray[byte], output: ptr byte,
     data = SSZ.decode(input, AttesterSlashingInput)
   except MalformedSszError, SszSizeMismatchError:
     let e = getCurrentException()
-    raise newException(FuzzCrashError, "SSZ deserialisation failed, likely bug in preprocessing.", e)
+    raise newException(
+      FuzzCrashError,
+      "SSZ deserialisation failed, likely bug in preprocessing.",
+      e,
+    )
 
   try:
     result = process_attester_slashing(data.state, data.attesterSlashing, cache)
-  except ValueError:
-    # TODO is a ValueError indicative of correct or incorrect processing code?
-    # If correct (but given invalid input), we should return false
-    # If incorrect, we should allow it to crash
-    result = false
+  except ValueError as e:
+    # TODO remove when status-im/nim-chronicles#60 is resolved
+    raise newException(
+      FuzzCrashError,
+      "Unexpected (logging?) error in attester slashing",
+      e,
+    )
 
   if result:
     result = copyState(data.state, output, output_size)
 
 
 proc nfuzz_block(input: openArray[byte], output: ptr byte,
-    output_size: ptr uint): bool {.exportc, raises:[FuzzCrashError, Defect].} =
+    output_size: ptr uint): bool {.exportc, raises: [FuzzCrashError, Defect].} =
   var data: BlockInput
 
   try:
     data = SSZ.decode(input, BlockInput)
   except MalformedSszError, SszSizeMismatchError:
     let e = getCurrentException()
-    raise newException(FuzzCrashError, "SSZ deserialisation failed, likely bug in preprocessing.", e)
+    raise newException(
+      FuzzCrashError,
+      "SSZ deserialisation failed, likely bug in preprocessing.",
+      e,
+    )
 
   try:
     result = state_transition(data.state, data.beaconBlock, {})
-  except IOError as e:
-    # TODO why an IOError?
-    raise newException(FuzzCrashError, "Unexpected IOError in state transition", e)
+  except IOError, ValueError:
```

Can't actually test with the raises pragma for `ValueError` without logging due to the general `Exception` that is thrown. Looks like `IOError` here is still raised without logging though.

---

All good, having it raise instead of return false will make it more visible to the fuzzer - if it gets triggered we can see what's happening.
If they aren't being caught at a higher level, it could still be thought of as an unexpected result/crash, and worth alerting on.